schema.go

```diff
@@ -231,9 +258,49 @@ func compare(t reflect.Type, src map[string]interface{}, opts *CompareOpts, resu
 				}
 
 				results.MismatchedFields = append(results.MismatchedFields, mismatch)
+				// There is no point to check nested fields if their parent is mismatched.
+				shouldCheckNested = false
 			}
 		} else {
-			results.MissingFields = append(results.MissingFields, fieldName)
+			missing := FieldMissing{Field: fieldName}
+
+			results.MissingFields = append(results.MissingFields, missing)
+			// There is no point to check nested fields if their parent is missing.
+			shouldCheckNested = false
+		}
+
+		if shouldCheckNested {
+			nested := src[fieldName].(map[string]interface{})
+			nestedType := f.Type
+			if f.Type.Kind() == reflect.Ptr {
+				nestedType = nestedType.Elem()
+			}
+
+			checkNestedFields(nestedType, fieldName, nested, opts, results)
+
+			continue
+		}
+	}
+}
+
+func checkNestedFields(t reflect.Type, fieldName string, src map[string]interface{}, opts *CompareOpts, results *CompareResults) {
+	// Remeber count of fields to check if new errors occured.
+	mismatchCount := len(results.MismatchedFields)
+	missingCount := len(results.MissingFields)
+
+	compare(t, src, opts, results)
+
+	// If there were new mismatched fields, add the current field name to their path.
+	if mismatchCount != len(results.MismatchedFields) {
+		for mi := mismatchCount; mi < len(results.MismatchedFields); mi++ {
+			results.MismatchedFields[mi].Path = append([]string{fieldName}, results.MismatchedFields[mi].Path...)
```

Comment:
Is this backwards?

```go
results.MismatchedFields[mi].Path = append(results.MismatchedFields[mi].Path, fieldName)
```

Reply:
We add parents on recursion ascend, so we have them in backwards order. I add them at the start of the Path to keep the Path in direct order
If the order in the Path doesn't matter I can change as you suggest